src/wobbly/models.py

```diff
@@ -1,20 +1,467 @@
+    ]
+
+    run_id: Annotated[
+        str | None,
+        Field(
+            title="Client-provided run ID",
+            description=(
+                "The run ID allows the client to add a unique identifier to"
+                " all jobs that are part of a single operation, which may aid"
+                " in tracing issues through a complex system or identifying"
+                " which operation a job is part of"
+            ),
+            examples=["daily-2024-10-29"],
+        ),
+    ] = None
+
+    destruction_time: Annotated[
+        datetime,
+        Field(
+            title="Destruction time",
+            description=(
+                "At this time, the job will be aborted if it is still"
+                " running, its results will be deleted, and it will either"
+                " change phase to ARCHIVED or all record of the job will be"
+                " discarded"
+            ),
+            examples=["2024-11-29T23:57:55+00:00"],
+        ),
+    ]
+
+    execution_duration: Annotated[
+        SecondsTimedelta | None,
+        Field(
+            title="Maximum execution duration",
+            description=(
+                "Allowed maximum execution duration. This is specified in"
+                " elapsed wall clock time (not CPU time). If null, the"
+                " execution time is unlimited. If the job runs for longer than"
+                " this time period, it will be aborted."
+            ),
+        ),
+        PlainSerializer(
+            lambda t: int(t.total_seconds()) if t is not None else None,
+            return_type=int,
+        ),
+    ] = None
+
+
+class JobCreate(JobBase):
+    """Information required to create a new UWS job."""
+
+
+class Job(JobBase):
+    """A single UWS job as stored in the UWS data store."""
+
+    id: Annotated[
+        str,
+        Field(
+            title="Job ID",
+            description="Unique identifier of the job",
+            examples=["47183"],
+        ),
+    ]
+
+    owner: Annotated[
+        str,
+        Field(
+            title="Job owner",
+            description="Identity of the owner of the job",
+            examples=["someuser"],
+        ),
+    ]
+
+    phase: Annotated[
+        ExecutionPhase,
+        Field(
+            title="Execution phase",
+            description="Current execution phase of the job",
+            examples=[
+                ExecutionPhase.PENDING,
+                ExecutionPhase.EXECUTING,
+                ExecutionPhase.COMPLETED,
+            ],
+        ),
+    ]
+
+    message_id: Annotated[
+        str | None,
+        Field(
+            title="Work queue message ID",
+            description=(
+                "Internal message identifier for the work queuing system."
+                " Only meaningful to the service that stored this ID."
+            ),
+            examples=["e621a175-e3bf-4a61-98d7-483cb5fb9ec2"],
+        ),
+    ] = None
+
+    creation_time: Annotated[
+        datetime,
```

Comment:
What's an `AwareDatetime`? I've never heard of that before. I only see a PyPI module with a 0.0.2 version that seems to have been abandoned.

I think if we wanted to add more validation here, we should force all datetimes to be UTC rather than just requiring they be time zone aware. I'll fix this to use `normalize_isodatetime` from Safir. I need to add a type wrapper in Safir to make that easy to use and will add that to my to-do list.

Reply:
https://docs.pydantic.dev/2.0/api/types/#pydantic.types.AwareDatetime
But yeah, `normalize_isodatetime` seems better.

Reply:
Oh, it's a Pydantic thing! I didn't think to check there and should have. TIL, that will be useful elsewhere.

Turned out that `normalize_datetime` was the better choice since using `normalize_isodatetime` rejected the default Python `.isoformat()` output, which seemed unnecessarily strict. That means we do accept times with a time zone, but at least they get immediately normalized to UTC.